packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -301,16 +302,25 @@ contract Deploy is Deployer {
     ///         2. The ProtocolVersions contract
     function setupSuperchain() public {
         console.log("Setting up Superchain");
+        DeploySuperchain deploySuperchain = new DeploySuperchain();
+        (DeploySuperchainInput dsi, DeploySuperchainOutput dso) = deploySuperchain.etchIOContracts();
 
-        // Deploy the SuperchainConfigProxy
-        deployERC1967ProxyWithOwner("SuperchainConfigProxy", mustGetAddress("SuperchainProxyAdmin"));
-        deploySuperchainConfig();
-        initializeSuperchainConfig();
+        // Set the input values on the input contract.
+        dsi.set(dsi.superchainProxyAdminOwner.selector, mustGetAddress("SuperchainProxyAdmin"));
+        // TODO: when DeployAuthSystem is done, finalSystemOwner should be replaced with the Foundation Upgrades Safe
+        dsi.set(dsi.protocolVersionsOwner.selector, cfg.finalSystemOwner());
+        dsi.set(dsi.guardian.selector, cfg.superchainConfigGuardian());
+        dsi.set(dsi.paused.selector, false);
 
-        // Deploy the ProtocolVersionsProxy
-        deployERC1967ProxyWithOwner("ProtocolVersionsProxy", mustGetAddress("SuperchainProxyAdmin"));
-        deployProtocolVersions();
-        initializeProtocolVersions();
+        dsi.set(dsi.requiredProtocolVersion.selector, ProtocolVersion.wrap(cfg.requiredProtocolVersion()));
+        dsi.set(dsi.recommendedProtocolVersion.selector, ProtocolVersion.wrap(cfg.recommendedProtocolVersion()));
+
+        // Run the deployment script.
+        deploySuperchain.run(dsi, dso);
+        save("SuperchainConfigProxy", address(dso.superchainConfigProxy()));
+        save("SuperchainConfig", address(dso.superchainConfigImpl()));
+        save("ProtocolVersionsProxy", address(dso.protocolVersionsProxy()));
+        save("ProtocolVersions", address(dso.protocolVersionsImpl()));
```

Comment:
Do we need to also save the SuperchainProxyAdmin output? I'm not sure if we use it, but is it useful to save for good measure? Deferring to you here, just wanted to suggest it

Reply:
I was trying to include the bare minimum of what we required here, but it may be helpful when it comes to adding in the Auth System, so doesn't hurt to include it here.